### PR TITLE
Add new authorize feature

### PR DESCRIPTION
**Pull request by EarthmanT**
* Adding the authorize another security group feature
* updating the existing tests to not break when we add this feature

### CODE REVIEW

File: ec2/securitygroup.py [hunk 1]
```diff
@@ -107,41 +105,78 @@ def delete(**_):
...
-                to_port=r['to_port'],
-                cidr_ip=r['cidr_ip'])
-        except (boto.exception.EC2ResponseError,
-                boto.exception.BotoServerError) as e:
-            raise NonRecoverableError('{0}'.format(str(e)))
-        except Exception as e:
-            _delete_security_group(group_id, ec2_client)
-            raise
+    for rule in ctx.node.properties['rules']:
+        if 'src_group_id' in rule:
```
Comment (ran-z) [inline ec2/securitygroup.py]: src_group_id and cidr_ip are mutually exclusive - the other 3 (protocol, from-port, to-port) are not

File: ec2/securitygroup.py [hunk 2]
```diff
@@ -107,41 +105,78 @@ def delete(**_):
...
+        _delete_security_group(group_object.id)
+        raise
+
+
+def _authorize_src_group(group_object, rule):
+
+    src_group_object = _get_security_group_from_id(rule['src_group_id'])
+
+    try:
+        group_object.authorize(src_group=src_group_object)
```
Comment (ran-z) [inline ec2/securitygroup.py]: need to pass the other params as well

File: plugin.yaml [hunk 3]
```diff
@@ -96,14 +95,13 @@ node_types:
       rules:
         default: []
         description: >
-          A list of rules, which are each a dictionary of key value pairs with the required
-          keys: ip_protocol (string), from_port (integer), to_port (integer), and 
-          cidr_ip (string).
+          You need to pass in either src_group_id OR ip_protocol, from_port, to_port,
```
Comment (ran-z) [inline plugin.yaml]: modify this too

File: ec2/securitygroup.py (hunk 1 above)
Comment (EarthmanT) [inline ec2/securitygroup.py]: made them mutually exclusive

File: ec2/securitygroup.py (hunk 2 above)
Comment (EarthmanT) [inline ec2/securitygroup.py]: did

File: ec2/securitygroup.py [hunk 4]
```diff
@@ -107,40 +105,55 @@ def delete(**_):
...
+
+            if 'cidr_ip' in rule:
+                raise NonRecoverableError(
+                    'You need to pass either src_group_id OR cidr_ip.')
+            src_group_object = \
+                _get_security_group_from_id(
+                    rule['src_group_id'])
+            del rule['src_group_id']
+            rule.update({'src_group': src_group_object})
+        elif 'cidr_ip' not in rule and 'src_group_id' not in rule:
```
Comment (ran-z) [inline ec2/securitygroup.py]: re-checking for src_group_id - pointless

**Pull request merged**